### ISSUE TITLE
[Ch 16][Notebook][Major] 체크포인트 Q3·해석 가이드가 지목한 "정치↔경제 혼동" 이 실측 1건 — "거의 없다"고 한 스포츠↔정치와 동수

### ISSUE THREAD

**Issue by jwrim9305** [yoon-gu/neuqes-101#179]
## 대상

- **챕터**: Ch 16 — 한국어 BERT Multi-class (KLUE-YNAT)
- **산출물**: Notebook (`16_ko_multiclass.ipynb` cell028 해석 가이드 · cell036 체크포인트 Q3), WikiDocs, 책
- **기준 커밋**: `bbae6f2` (master)
- **환경**: Colab T4, 챕터와 동일 셋업(5K/1K · 2 epoch · lr 2e-5 · bs 16 · max_length 128)으로 **시드 3개**(42·1234·2026) 학습 후 혼동 행렬 집계. 노트북은 혼동 행렬을 그림으로만 출력해 숫자를 직접 볼 수 없어 별도로 집계했습니다.

## 실제 결과

체크포인트 Q3 가 이렇게 묻습니다.

> 3. 혼동 행렬에서 ***정치 ↔ 경제* 혼동이 많은데 *스포츠 ↔ 정치* 혼동은 거의 없는** 이유는?

해석 가이드(cell028)도 같은 전제를 씁니다.

> - **오답 패턴**:
>   - **정치 ↔ 경제**: 둘 다 정책·법안·국제 이슈 다뤄 *경계가 모호* — 자연스러운 혼동
>   - 생활문화 ↔ 사회: 사회 이슈 vs 일상·문화 보도 — 헤드라인 한 줄로는 사람도 헷갈리는 경계
>   - IT과학 ↔ 경제: 기업·산업 뉴스가 양쪽에 걸침
> - **먼 클래스 혼동** (스포츠 ↔ 정치 등) 이 자주 보이면 라벨 노이즈나 학습 부족 신호.

**실측 혼동 행렬입니다** (행=실제, 열=예측, eval 1,000건).

```
               IT과학       경제       사회     생활문화       세계      스포츠       정치
      IT과학       56        0        2        0        0        0        0
        경제        7      133       12        2        3        0        0
        사회        7       24      331       21        3        1       13
    생활문화        2        2       13      127        0        2        0
        세계        0        1        2        3       89        1        1
      스포츠        0        0        2        0        1       70        1
        정치        0        1       14        0        2        0       51
```

챕터가 지목한 쌍을 뽑으면 이렇습니다.

| 쌍 | 챕터 서술 | seed 42 | seed 1234 | seed 2026 | 판정 |
|---|---|---|---|---|---|
| **정치 ↔ 경제** | 「혼동이 많다」 | **1** | **1** | **1** | ❌ |
| 생활문화 ↔ 사회 | 「사람도 헷갈리는 경계」 | 34 | 32 | 31 | ✅ |
| **스포츠 ↔ 정치** | 「거의 없다」 | 1 | 1 | **2** | ✅ |

**「많다」고 한 쌍이 세 시드 모두 1건**이고, 「거의 없다」고 한 쌍과 같거나 오히려 적습니다(seed 2026 에서는 1 < 2 로 역전). 체크포인트 Q3 는 존재하지 않는 차이를 설명하라고 요구합니다.

`IT과학 ↔ 경제`(「기업·산업 뉴스가 양쪽에 걸침」)는 seed 42 에서 7건인데 **전부 경제 → IT과학 한 방향**이었습니다. 「양쪽에 걸친다」는 서술과 맞지 않습니다.

## 실제 혼동은 다른 곳에 몰려 있습니다

```
seed 42    사회→경제 24, 사회→생활문화 21, 정치→사회 14, 생활문화→사회 13
seed 1234  사회→경제 29, 사회→생활문화 20, 정치→사회 14, 생활문화→사회 12
seed 2026  사회→경제 23, 생활문화→사회 18, 경제→사회 16, 사회→정치 14
```

**세 시드 모두 상위 쌍 전부에 `사회` 가 한쪽으로 들어 있습니다.** `사회` 가 eval 1,000건 중 400건(40%)으로 압도적 다수 클래스라, 모델이 애매하면 `사회` 쪽으로 밀거나 `사회` 를 다른 클래스로 흘립니다.

챕터가 짚어야 할 패턴은 **「다수 클래스 쏠림」** 인데, 서술은 **「의미상 가까운 카테고리끼리 헷갈린다」** 로 되어 있습니다. 설명의 축이 다릅니다.

> `생활문화 ↔ 사회`(34건)만 두 설명이 겹칩니다 — 의미도 가깝고 `사회` 도 끼어 있어서입니다.

## 덧붙여 — 「모든 셀이 0.85+」 기준도 안 맞습니다

> - **대각선 셀** = 그 카테고리의 재현율. **모든 셀이 0.85+ 면 잘 학습된 것.**

실측 재현율입니다.

| 카테고리 | seed 42 | seed 1234 | seed 2026 | 0.85 |
|---|---|---|---|---|
| IT과학 | 0.966 | 0.948 | 0.914 | ✅ |
| 경제 | 0.847 | 0.879 | 0.809 | △ |
| 사회 | 0.828 | 0.832 | 0.855 | △ |
| 생활문화 | 0.870 | 0.877 | 0.829 | △ |
| 세계 | 0.918 | 0.907 | 0.907 | ✅ |
| 스포츠 | 0.946 | 0.959 | 0.946 | ✅ |
| **정치** | **0.750** | **0.750** | **0.794** | ❌ |

**「모든 셀이 0.85+」를 만족하는 시드가 하나도 없습니다.** 특히 `정치` 는 세 시드 모두 0.79 이하로, 시드 문제가 아니라 표본이 68건뿐인 구조 문제입니다. 챕터는 이 결과를 「7클래스 분류치고 견고합니다」라고 서술합니다.

## 기대 결과

**실측에 맞춰 오답 패턴을 다시 씁니다.** 인과 설명의 축을 「의미 유사성」에서 「다수 클래스 쏠림 + 의미 유사성」으로 넓히는 방향입니다.

```diff
  - **오답 패턴**:
-   - 정치 ↔ 경제: 둘 다 정책·법안·국제 이슈 다뤄 *경계가 모호* — 자연스러운 혼동
-   - 생활문화 ↔ 사회: 사회 이슈 vs 일상·문화 보도 — 헤드라인 한 줄로는 사람도 헷갈리는 경계
-   - IT과학 ↔ 경제: 기업·산업 뉴스가 양쪽에 걸침 (예: "삼성전자 4분기 실적 발표")
+   - **다수 클래스로 쏠림** — eval 에서 *사회* 가 가장 큰 비중을 차지해, 애매한 헤드라인이
+     *사회* 로 흘러가거나 *사회* 가 다른 카테고리로 새는 쌍이 오답의 대부분입니다.
+     위 행렬에서 오답 상위 쌍을 확인해 보세요.
+   - **의미가 가까운 경계** — 생활문화 ↔ 사회처럼 일상·문화 보도와 사회 이슈는 헤드라인
+     한 줄로는 사람도 헷갈립니다.
+   - **비대칭 혼동** — 경제 → IT과학처럼 한 방향으로만 새는 쌍도 있습니다 (기업·산업
+     뉴스가 양쪽에 걸치지만 학습된 시그널 단어가 한쪽에 쏠린 경우). FAQ Q2 참고.
  - **먼 클래스 혼동** (스포츠 ↔ 정치 등) 이 자주 보이면 라벨 노이즈나 학습 부족 신호.
```

체크포인트 Q3 는 실제로 관찰되는 현상을 묻도록 바꿉니다.

```diff
- 3. 혼동 행렬에서 *정치 ↔ 경제* 혼동이 많은데 *스포츠 ↔ 정치* 혼동은 거의 없는 이유는?
+ 3. 혼동 행렬에서 오답이 가장 많은 쌍들의 공통점은 무엇인가요? 클래스별 표본 수와 함께
+    설명해 보세요.
```

「모든 셀이 0.85+」 기준도 실측과 맞추거나(예: 0.75+), 「클래스별 표본 수가 적을수록 재현율이 흔들린다」는 단서를 붙이는 편이 낫습니다.

> **수치를 산문에 박지 않았습니다.** 「위 행렬에서 확인해 보세요」로 산출물을 가리키게 썼습니다 — 재실행하면 건수가 달라지지만 「다수 클래스로 쏠린다」는 구조는 유지됩니다.

## 재현 절차

```python
# Colab T4. 챕터와 동일 셋업으로 학습 후
from sklearn.metrics import confusion_matrix
cm = confusion_matrix(labels, preds, labels=list(range(7)))
pairs = [(cm[i][j], LABEL_NAMES[i], LABEL_NAMES[j])
         for i in range(7) for j in range(7) if i != j]
for v, a, b in sorted(pairs, reverse=True)[:6]:
    print(f"{a} → {b}: {v}")
# 사회→경제 24 · 사회→생활문화 21 · 정치→사회 14 · 생활문화→사회 13 · 사회→정치 13 · 경제→사회 12
```

시드만 바꿔(42·1234·2026) 세 번 반복했습니다.

## 전파 범위

| 산출물 | 수정 | 재실행/재생성 |
|---|---|---|
| 📓 Notebook | `cell028` 해석 가이드 · `cell036` 체크포인트 Q3 | **재실행 불필요** — markdown 셀만 변경 |
| 📄 WikiDocs | `pages/16-ko_multiclass-anatomy.md` · `-wrapup.md` | 재변환 반영 |
| 📕 EPUB(tex) | 재생성 | 재생성 필요 |

## 심각도

**Major 로 제안합니다.**

- **체크포인트 질문이 존재하지 않는 현상을 설명하라고 요구합니다.** 성실한 독자는 혼동 행렬을 들여다보며 「정치↔경제 혼동」을 찾다가 못 찾습니다.
- 해석 가이드가 제시한 인과 설명이 **자기 데이터로 반증**됩니다 — 세 쌍 중 하나만 맞습니다.
- 「모든 셀이 0.85+」 기준도 자기 결과가 미달인데 「견고하다」고 서술합니다.
- 수치 오기가 아니라 **가르치는 내용이 틀린** 유형이라 Ch 13 [#107](https://github.com/yoon-gu/neuqes-101/issues/107) · Ch 14 [#165](https://github.com/yoon-gu/neuqes-101/issues/165) 와 등급을 맞췄습니다.

> **시드 3개로 확인했습니다.** eval 셋은 `shuffle(seed=42)` 로 고정이라 시드와 무관하게 같고, 바뀌는 것은 모델 초기값뿐입니다. `정치↔경제` 는 세 번 모두 1건, 상위 오답은 세 번 모두 `사회` 쏠림, `정치` recall 은 세 번 모두 0.79 이하였습니다. 「1회 실측이라 우연일 수 있다」는 반론은 해소됩니다.
>
> eval 클래스 분포: 사회 400 · 경제 157 · 생활문화 146 · 세계 97 · 스포츠 74 · 정치 68 · IT과학 58

## 담당

- 수정 담당자: 미정
- 재검수자: 등록자 (@jwrim9305)
- 연결: 챕터 검수 이슈 [#40](https://github.com/yoon-gu/neuqes-101/issues/40) / 동일 유형 [#107](https://github.com/yoon-gu/neuqes-101/issues/107) · [#165](https://github.com/yoon-gu/neuqes-101/issues/165) / 관련 [#157](https://github.com/yoon-gu/neuqes-101/issues/157)(혼동 행렬 라벨 표기)

**Comment by jwrim9305** [comment]
수정했습니다. 브랜치: `fix/ch16-review-2026-09-b1` · `cf26ea3`

### 해석 가이드 (`cell028`)

```diff
- - **대각선 셀** = 그 카테고리의 재현율. 모든 셀이 0.85+ 면 잘 학습된 것.
+ - **대각선 셀** = 그 카테고리의 재현율. 표본이 많은 카테고리일수록 안정적이고, 표본이
+   적은 카테고리는 흔들립니다 — 위 행렬을 클래스별 support 와 함께 보세요.
  - **오답 패턴**:
-   - 정치 ↔ 경제: 둘 다 정책·법안·국제 이슈 다뤄 경계가 모호 — 자연스러운 혼동
-   - 생활문화 ↔ 사회: 사회 이슈 vs 일상·문화 보도 — 헤드라인 한 줄로는 사람도 헷갈리는 경계
-   - IT과학 ↔ 경제: 기업·산업 뉴스가 양쪽에 걸침 (예: "삼성전자 4분기 실적 발표")
+   - **다수 클래스로 쏠림** — eval 에서 *사회* 가 가장 큰 비중을 차지해, 애매한 헤드라인이
+     *사회* 로 흘러가거나 *사회* 가 다른 카테고리로 새는 쌍이 오답의 대부분입니다.
+     위 행렬에서 오답 상위 쌍을 확인해 보세요.
+   - **의미가 가까운 경계** — 생활문화 ↔ 사회처럼 일상·문화 보도와 사회 이슈는 헤드라인
+     한 줄로는 사람도 헷갈립니다.
+   - **비대칭 혼동** — 한 방향으로만 새는 쌍도 있습니다 (기업·산업 뉴스가 양쪽에 걸치지만
+     학습된 시그널 단어가 한쪽에 쏠린 경우). FAQ Q2 참고.
```

**틀린 쌍(정치↔경제)만 걷어내고 맞는 쌍(생활문화↔사회)은 남겼습니다.** 「IT과학 ↔ 경제」는 실측이 한 방향(경제→IT과학)뿐이라 방향을 특정하지 않고 「비대칭 혼동」 항목으로 옮겼습니다 — FAQ Q2 가 이미 대칭/비대칭을 설명하고 있어 그쪽으로 연결했습니다.

### 체크포인트 Q3 (`cell036`)

```diff
- 3. 혼동 행렬에서 *정치 ↔ 경제* 혼동이 많은데 *스포츠 ↔ 정치* 혼동은 거의 없는 이유는?
+ 3. 혼동 행렬에서 오답이 가장 많은 쌍들의 공통점은 무엇인가요? 클래스별 표본 수와 함께
+    설명해 보세요.
```

전제가 성립하지 않아 문구 수정으로는 해결되지 않는 자리라 **질문을 교체**했습니다. 바꾼 질문은 실제 데이터로 답이 나오고, 「불균형 데이터에서 모델이 다수 클래스로 쏠린다」는 더 일반적인 교훈으로 이어집니다.

### 재현율 기준

`정치` recall 이 세 시드 모두 0.79 이하(0.750·0.750·0.794)라 **0.85 기준을 만족하는 시드가 없었습니다.** 표본이 68건뿐인 구조 문제라, 일률 기준 대신 **표본 수와 함께 보라**는 안내로 바꿨습니다.

| 산출물 | 반영 | 재실행/재생성 |
|---|---|---|
| 📓 노트북 | ✅ `cell028` · `cell036` | **재실행 불필요** — markdown 셀만 변경 |
| 📄 WikiDocs | ✅ `anatomy.md` · `wrapup.md` | 재변환 불필요 (직접 반영함) |
| 📕 EPUB(tex) | ⏳ 머지 후 | 재생성 필요 |

수치는 산문에 넣지 않고 **「위 행렬에서 확인해 보세요」** 로 산출물을 가리키게 썼습니다. 재실행으로 건수가 달라져도 「다수 클래스로 쏠린다」는 구조는 유지됩니다.

**Comment by sy2399** [comment]
재검수 확인 (@sy2399) — `1f85690` 기준. 해석 가이드(c28)·체크포인트 Q3(c36)가 "정치↔경제 vs 스포츠↔정치" 전제를 버리고 표본 수·다수 클래스 쏠림 프레임으로 재작성된 것 확인. 새 실행본에서도 정치 recall 0.750(표본 68건)으로 재작성 서술과 부합, "모든 셀이 0.85+" 기준 제거 확인. ④ 라벨별 해석 산문을 같은 프레임으로 재작성했습니다. pages·tex·EPUB 전파 확인(옛 서술 잔여 0). PR #226 병합 시 종결됩니다.

**Comment by sy2399** [comment]
PR #226 병합이 revert 되어(`bfa2459`) 수정이 master 에서 내려갔습니다. 수정본은 `fix/ch16-review-2026-09-merged` 브랜치에 유지 — 재병합 시 다시 종결됩니다.
